src/units.cpp

```diff
@@ -464,6 +464,91 @@ double Units::scalingFactor(const UnitsPtr &units1, const UnitsPtr &units2)
     return 0.0;
 }
 
+using unitsMap = std::map<std::string, double>;
+
+void updateUnitMap(const UnitsPtr &units, unitsMap &unitMap, double exp = 1.0)
+{
+    if (units->isBaseUnit()) {
+        auto found = unitMap.find(units->name());
+        if (found == unitMap.end()) {
+            unitMap.emplace(units->name(), 1.0);
+        } else {
+            auto unit = unitMap.find(units->name());
+            unit->second += exp;
+        }
+    } else {
+        for (size_t i = 0; i < units->unitCount(); ++i) {
+            std::string ref;
+            std::string pre;
+            std::string id;
+            double expMult;
+            units->unitAttributes(i, ref, pre, exp, expMult, id);
+            if (isStandardUnitName(ref)) {
+                auto unit = standardUnitsList.find(ref);
+                for (const auto &u : unit->second) {
+                    if (unitMap.find(u.first) == unitMap.end()) {
+                        unitMap.emplace(u.first, u.second * exp);
+                    } else {
+                        auto ut = unitMap.find(u.first);
+                        ut->second += u.second * exp;
+                    }
+                }
+            } else {
+                auto model = owningModel(units);
+                if (model != nullptr) {
+                    auto refUnits = model->units(ref);
+                    if ((refUnits == nullptr) || refUnits->isImport()) {
+                        unitMap.clear();
+                        break;
+                    }
+                    updateUnitMap(refUnits, unitMap, exp); // recursively add units to the map
+                }
+            }
+        }
+    }
+}
+
+bool Units::isEquivalentTo(const UnitsPtr &units1, const UnitsPtr &units2)
+{
+    // initial checks
+    if ((units1 == nullptr) || (units2 == nullptr)) {
+        return false;
+    }
+    if ((units1->isImport()) || (units2->isImport())) {
+        return false;
+    }
+
+    unitsMap units1Map;
+    updateUnitMap(units1, units1Map);
+    unitsMap units2Map;
+    updateUnitMap(units2, units2Map); // creating the maps to compare units over
+
+    if (units1Map.size() == units2Map.size()) {
+        auto it = units1Map.begin();
+
+        while (it != units1Map.end()) {
+            std::string unit = it->first;
+            auto found = units2Map.find(unit);
+
+            if (found == units2Map.end()) {
+                return false;
+            }
+            if ((found->second < it->second) || (found->second > it->second)) {
+                return false;
+            }
+
+            it++;
```

Comment:
`++it;` is generally preferred (I think).

Reply:
Indeed, it's preferred, I removed it in the end. :)